charts/argo-rollouts/templates/controller/deployment.yaml

```diff
@@ -17,8 +17,13 @@ spec:
     matchLabels:
       app.kubernetes.io/component: {{ .Values.controller.component }}
       {{- include "argo-rollouts.selectorLabels" . | nindent 6 }}
+  {{- if eq .Values.controller.deploymentStrategy.type "RollingUpdate" }}
+  strategy:
+    {{- toYaml .Values.controller.deploymentStrategy | nindent 4 }}
+  {{- else }}
   strategy:
     type: Recreate
+  {{- end }}
```

Comment:
Hi @nedal87 , thank you for your PR.

Did you investigate why this is hardcoded as `Recreate` originally ? I am not sure about this case, but some resources are not designed for `RollingUpdate` .
If the upstream's original kustomize is `RollingUpdate` , then it sounds safe to provide other choices but [upstream's manifest](https://github.com/argoproj/argo-rollouts/blob/master/manifests/base/argo-rollouts-deployment.yaml#L50) is `Recreate`

Can you please check it in upstream?

Reply:
Not sure for now but at the point of March 2019, it should be `Recreate` . 👀 
https://github.com/argoproj/argo-rollouts/issues/38

Reply:
>Are there any specific concerns regarding the usage of RollingUpdate deployment strategy?

Aside from the ISSUE above, I don't have any concerns.
Since we follow upstream's manifest and not familiar with the detail of feature, can you please check in upstream for in case?

Reply:
Thank you for your feedback, I will reach out to the upstream to verify and proceed from there.

Reply:
Thank you! When we are sure about this, LGTM 🚀

Reply:
Hi @nedal87 ,
I investigated and revealed that controller can be RollingUpdate surely, so I made PR to upstream (it's already merged into master).
https://github.com/argoproj/argo-rollouts/pull/3334

When upstream releases the PR as specific version, let's change the default type as `RollingUpdate` in this PR.

Reply:
Hi @yu-croco that's great news. Thanks a lot! Updating the PR to have the `RollingUpdate` by default.